disco_aws_automation/disco_vpc.py

```diff
@@ -484,111 +286,103 @@ def _configure_environment(self):
                                      % self.environment_name)
 
             # get the cidr for all other VPCs so we can avoid overlapping with other VPCs
-            occupied_network_cidrs = [vpc.cidr_block for vpc in self.list_vpcs()]
+            occupied_network_cidrs = [vpc['cidr_block'] for vpc in self.list_vpcs()]
 
             vpc_cidr = DiscoVPC.get_random_free_subnet(ip_space, int(vpc_size), occupied_network_cidrs)
 
             if vpc_cidr is None:
                 raise VPCConfigError('Cannot create VPC %s. No subnets available' % self.environment_name)
 
         # Create VPC
-        vpc_conn = VPCConnection()
-        self.vpc = vpc_conn.create_vpc(vpc_cidr)
-        keep_trying(300, self.vpc.add_tag, "Name", self.environment_name)
-        keep_trying(300, self.vpc.add_tag, "type", self.environment_type)
+        self.vpc = self.boto3_ec2.create_vpc(CidrBlock=str(vpc_cidr))['Vpc']
+        waiter = self.boto3_ec2.get_waiter('vpc_available')
+        waiter.wait(VpcIds=[self.vpc['VpcId']])
+        ec2 = boto3.resource('ec2')
+        vpc = ec2.Vpc(self.vpc['VpcId'])
+        tags = vpc.create_tags(Tags=[{'Key': 'Name', 'Value': self.environment_name},
+                                     {'Key': 'type', 'Value': self.environment_type}])
         logging.debug("vpc: %s", self.vpc)
+        logging.debug("vpc tags: %s", tags)
 
-        # Create metanetworks (subnets, route_tables and security groups)
-        self._networks = self._create_new_meta_networks()
-        for network in self.networks.values():
-            network.create()
-
-        dhcp_options = self._configure_dhcp()
-        self.vpc.connection.associate_dhcp_options(dhcp_options.id, self.vpc.id)
+        dhcp_options = self._configure_dhcp()[0]
```

Comment:
It should always be != 0, unless something wrong on AWS side. Actually `self._configure_dhcp()` should just return the single instance of `dhcp_options`. I'll update it.